File: seed/php-sdk/enum/src/Queryparam/Requests/QueryParamSendRequest.php
<?php

namespace Seed\Queryparam\Requests;

use Seed\Core\Json\JsonSerializableType;
use Seed\Types\Operand;
use Seed\Types\Color;

class QueryParamSendRequest extends JsonSerializableType
{
    /**
     * @var value-of<Operand> $operand
     */
    public string $operand;

    /**
     * @var ?value-of<Operand> $maybeOperand
     */
    public ?string $maybeOperand;

    /**
     * @var value-of<Color> $operandOrColor
     */
    public string $operandOrColor;

    /**
     * @var (
     *    value-of<Color>
     *   |value-of<Operand>
     * )|null $maybeOperandOrColor
     */
    public string|null $maybeOperandOrColor;

    /**
     * @param array{
     *   operand: value-of<Operand>,
     *   operandOrColor: value-of<Color>,
     *   maybeOperand?: ?value-of<Operand>,
     *   maybeOperandOrColor?: (
     *    value-of<Color>
     *   |value-of<Operand>
     * )|null,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->operand = $values['operand'];
        $this->maybeOperand = $values['maybeOperand'] ?? null;
        $this->operandOrColor = $values['operandOrColor'];
        $this->maybeOperandOrColor = $values['maybeOperandOrColor'] ?? null;
    }
}

File: seed/csharp-sdk/enum/plain-enums/src/SeedApi/Queryparam/Requests/QueryParamSendRequest.cs
using global::System.Text.Json.Serialization;
using OneOf;
using SeedApi.Core;

namespace SeedApi;

[Serializable]
public record QueryParamSendRequest
{
    [JsonIgnore]
    public required Operand Operand { get; set; }

    [JsonIgnore]
    public Operand? MaybeOperand { get; set; }

    [JsonIgnore]
    public required Color OperandOrColor { get; set; }

    [JsonIgnore]
    public OneOf<Color, Operand>? MaybeOperandOrColor { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
